codice/ddf
# [WIP] DDF-4072 Run solr inside of a docker container during itests

#### What does this PR do?

This PR introduces some changes to the way itests are run with regards to the external solr process.

With this change solr is now run inside of a docker container instead of manually unpacked and started in the same environment as ddf.

This is the first of a few changes I am proposing for the current itest process which will simply replace the current single solr instance with one that is run inside of a separate container. The next change I will propose, assuming this is merged, will be to have a clean solr instance per test as this should provide a faster clean-up than clearing the single solr instance after each test. This latter change will require more experimentation however and has been left out for now.

Additional Details:

The `distribution/docker/solr` module creates a ddf compatible solr image by building on top of the official solr image and adding in a few things:
* Adds jts-core
* Adds solr-xpath
* Adds a bundled copy of the ddf solr configuration files
* Adds a helper script for creating new ddf solr cores
* Adds pre-created cores for `catalog`, `metacard_cache`, `alerts`, `preferences`, `subscriptions`

#### Who is reviewing it? 

@ahoffer @mcalcote

#### Select relevant component teams:
 
@codice/build @codice/test 

#### Ask 2 committers to review/merge the PR and tag them here.

@rzwiefel @pklinef 

#### How should this be tested?

 1. run a quickbuild
 2. ensure that `docker image ls` shows `codice/ddf-solr 2.14.0-SNAPSHOT` was created
 3. run the itests

#### Any background context you want to provide?

There were several drivers behind this proposed change:
* Inconsistent execution of solr across all environments
* Encountering issues with missing binaries used by solr start scripts
* Encountering issues with certain characters in folder paths automatically used by CI that were breaking solr start scripts
* Occasionally itests would start before solr was ready

#### What are the relevant tickets?
[DDF-4072](https://codice.atlassian.net/browse/DDF-4072)
#### Screenshots

#### Checklist:
- [ ] Update any itests that won't work automatically with this change

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
